### DISCUSSION TITLE
Delay in libre3 values

### DISCUSSION THREAD

**Discussion by enpa23** [NightscoutFoundation/xDrip#2598] [General]
Hi, moving to FSL3 has forced me to use Juggluco as data source for xDrip ( hardware data source in settings: libre 2 patched app ), as it is the only way to connect to the new abbt sensor other than relying on an internet upload with all its issues.

Juggluco is locally mirroring data from an official patched app or connecting directly to the sensor, either way its data is already processed by the same abbt algorithm.
xDrip adds then a second layer of smoothing/averaging which leads to a big delay in actual data in case of rapidly changing BG values, a delay that added to the natural delay of interstitial fluid could be easily critical for treatment choices.
 I cannot figure a way to disable or at least reduce it, if there's any.
A falling example:
![image](https://user-images.githubusercontent.com/45668841/210425699-4ad7d821-d90b-4777-b012-fbd5cbe018ae.png)
![image](https://user-images.githubusercontent.com/45668841/210425739-0d4849f4-780c-45ab-b432-271a35b1c3c1.png)
Here a rising example:
![image](https://user-images.githubusercontent.com/45668841/210425821-7c83054c-7d29-474e-86e3-35bc613a6684.png)

It was told me that xDrip uses data from the last 20 minutes for averaging purposes, reducing to 5 minutes or so would probably be enough with this setting.
I tried also the @jakobheyman version ( xDripJH ) which can display raw values ( in this case they are in fact not raw at all ) but it is bounded to the one minute update approach which makes me impossible to use watchdrip as it leads to constant watchface uploads. Also I'm not sure about its trends data because a minute time delta would lead to a non significant bg delta.

**Comment by jakobheyman** [comment]
I believe that when I was using the patched Libre 2 app (perhaps two years ago) I got rid of the delay if using Datricsae calibration under Settings > Less common settings > Advanced Calibration > Calibration Plugin, plus setting "Use Plugin Glucose" to ON. I might be wrong but I think it was because xDrip then (with Datricsae calibration) always used only the last raw value to get the calibrated bg value.

**Reply by enpa23** [reply]
Apparently it does not work
![image](https://user-images.githubusercontent.com/45668841/210601929-b9141025-4691-4e23-bd48-5d70039a0e6d.png)

Latest main data point has a lower value than the last 9 raw data points, so it's clearly averaging with some other older data.

![image](https://user-images.githubusercontent.com/45668841/210602851-32b7aac4-c97f-415f-8d17-3089470a6235.png)

**Comment by jakobheyman** [comment]
Sorry for that!
Else I don't know how to avoid the delay.

**Reply by enpa23** [reply]
I tried to rebuild the app using this patch from your version to get raw data as main, but apparently it does not work.
https://github.com/jakobheyman/xdripJH/commit/9e39682761488febb914d872ddbe06106132aed7

Does it need a specific setting on xDrip or do you think that that I have to patch also other parts of the code?

**Comment by jakobheyman** [comment]
That patch does not change anything with the delay or smoothing. I guess we should not discuss my modified version of xDrip here and I have therefore set up a Discussions page for my fork where you can write if you wish.
